### DISCUSSION TITLE
Feature Request: pass Event through to `onChange`

### DISCUSSION THREAD

**Discussion by Nokel81** [JedWatson/react-select#4650] [Ideas]
## Description

Pass the `MouseEvent` or `KeyboardEvent` as the third parameter of `onChange`

## Use Case

I would like to implement a form of a multi-select component that acts as `isMulti={false}` normally but acts like `isMulti={true}` when the `"SHIFT"` key is held during a click.

## Interface

This seems like a rather niche feature so allowing the user to do this instead of a new prop seems the best course of action.

## Version

I am currently using v3 but from the docs this doesn't look like it would be possible in v4 either.

**Comment by ebonow** [comment]
Greetings @Nokel81 ,

I believe I understand what you are trying to achieve, though I believe passing a `MouseEvent` or `KeyboardEvent` in all of the action events might be overkill and potentially confusing. 

Instead, I believe it would be easier to use a `keyDown/keyUp` listener for the `"SHIFT"` key.

Demo: [codesandbox](https://codesandbox.io/s/react-select-onchange-shift-i9prr)
This seems to be a working solution though you likely have other use-cases that I didn't test.

Admittedly, this solution would be even cleaner without the need for a custom Input component if `onKeyUp` was supported as a prop in the same way that `onKeyDown` is.

Please let me know if this satisfies both your use-case and request for passing the native event.

**Comment by Nokel81** [comment]
Yes this satisfies my use-case, though as you said having `onKeyUp` would be a better solution.
